File: frontend/app/models/dto/Extrato.ts
export type TipoTransacao = "depósito" | "saque" | "transferência"

export type Movimentacao = {
  id: number
  data: string
  tipo: TipoTransacao
  origem: string | null
  destino: string | null
  valor: string
  }

  export type Extrato = {
    conta: string
    saldo: number
    movimentacoes: Movimentacao[]
  }

File: backend/ms-conta/src/main/java/com/ufpr/bantads/conta/application/dto/response/Extrato.java
package com.ufpr.bantads.conta.application.dto.response;

import com.ufpr.bantads.conta.domain.model.entity.MovimentacaoQuery;

public record Extrato(
    String data,
    String tipo,
    String origem,
    String destino,
    Double valor
) {
    public static Extrato fromEntity(MovimentacaoQuery movimentacaoQuery) {
        return new Extrato(
            movimentacaoQuery.getDataHora().toString(),
            movimentacaoQuery.getTipo().name(),
            movimentacaoQuery.getContaOrigemNumero(),
            movimentacaoQuery.getContaDestinoNumero(),
            movimentacaoQuery.getValor().doubleValue()
        );
    }
}
